infra/pr_helper.py

```diff
@@ -0,0 +1,232 @@
+    self._pr_author = os.environ['PRAUTHOR']
+    self._token = os.environ['GITHUBTOKEN']
+    self._pr_number = os.environ['PRNUMBER']
+    self._headers = {
+        'Authorization': f'Bearer {self._token}',
+        'X-GitHub-Api-Version': '2022-11-28'
+    }
+    os.environ['GITHUB_AUTH_TOKEN'] = self._token
+
+  def get_pr_author(self):
+    """Gets the pr author user name."""
+    return self._pr_author
+
+  def get_projects_path(self):
+    """Gets the current project path."""
+    response = requests.get(f'{BASE_URL}/pulls/{self._pr_number}/files',
+                            headers=self._headers)
+
+    projects_path = set()
+    for file in response.json():
+      file_path = file['filename']
+      dir_path = os.path.dirname(file_path)
+      if dir_path is not None and dir_path.split('/')[0] == 'projects':
+        projects_path.add(dir_path)
+    return list(projects_path)
+
+  def get_author_email(self):
+    """Retrieves the author's email address for a pull request,
+    including non-public emails."""
+    commits_response = requests.get(
+        f'{BASE_URL}/pulls/{self._pr_number}/commits', headers=self._headers)
+    if not commits_response.ok:
+      return None
+    email = commits_response.json()[0]['commit']['author']['email']
+    return email
+
+  def get_project_yaml(self, project_path):
+    """Gets the project yaml file."""
+    contents_url = f'{BASE_URL}/contents/{project_path}/project.yaml'
+    return self.get_yaml_file_content(contents_url)
+
+  def get_yaml_file_content(self, contents_url):
+    """Gets yaml file content."""
+    response = requests.get(contents_url, headers=self._headers)
+    if not response.ok:
+      return {}
+    content = base64.b64decode(response.json()['content']).decode('UTF-8')
+    return yaml.safe_load(content)
+
+  def get_integrated_project_info(self):
+    """Gets the new integrated project."""
+    response = requests.get(f'{BASE_URL}/pulls/{self._pr_number}/files',
+                            headers=self._headers)
+
+    for file in response.json():
+      file_path = file['filename']
+      if 'project.yaml' in file_path:
+        return self.get_yaml_file_content(file['contents_url'])
+
+    return None
+
+  def get_pull_request_url(self, commit):
+    """Gets the pull request url."""
+    pr_response = requests.get(f'{BASE_URL}/commits/{commit}/pulls',
+                               headers=self._headers)
+    if not pr_response.ok:
+      return None
+    return pr_response.json()[0]['html_url']
+
+  def is_author_internal_member(self):
+    """Returns if the author is an internal member."""
+    response = requests.get(f'{BASE_URL}/contents/infra/MAINTAINERS.csv',
+                            headers=self._headers)
+    if not response.ok:
+      return False
+
+    maintainers = base64.b64decode(response.json()['content']).decode('UTF-8')
+    for line in maintainers.split('\n'):
+      if self._pr_author == line.split(',')[2]:
+        save_env(None, None, True)
```

Comment:
nit: this is a side effect that seems surprising based on the function name, which is just returning a boolean. We should move this logic to the caller.